pkg/controllers/core/deployer_controller.go

```diff
@@ -132,6 +138,14 @@ func (r *DeployerReconciler) reconcile(ctx context.Context, log logr.Logger, dep
 	deployer.Status.ServiceName = childService.Name
 	deployer.Status.PropagateServiceStatus(&childService.Status)
 
+	ingress, err := r.reconcileIngress(ctx, log, deployer, childService.Name)
+	if err != nil {
+		log.Error(err, "unable to reconcile Ingress", "deployer", deployer)
+		return ctrl.Result{}, err
+	}
+
+	deployer.Status.IngressName = ingress.Name
```

Comment:
The status of the ingress should be reflected on the deployer, and contribute to the deployer's ready condition

```suggestion
	deployer.Status.IngressName = ingress.Name
	deployer.Status.PropagateIngressStatus(&childIngress.Status)
```